pkg/controller/tektoninstallation/tektoninstallation_controller.go

```diff
@@ -87,41 +107,129 @@ func (r *ReconcileTektonInstallation) Reconcile(request reconcile.Request) (reco
 		return reconcile.Result{}, err
 	}
 
-	err := r.EnsureTektonSubscription(reqLogger, tektonInstallation)
-	return reconcile.Result{}, err
+	if created, err := r.EnsureTektonSubscription(reqLogger, tektonInstallation); err != nil {
+		return reconcile.Result{}, err
+	} else if created {
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonSubscriptionCreated, "created tekton subscription")
+	}
+
+	if requeue, err := r.ensureWatchTektonCluster(); err != nil {
+		return reconcile.Result{}, r.wrapErrorWithStatusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, err, "failed to add watch for TektonCluster")
+	} else if requeue {
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
+	}
+
+	cluster := &config.Config{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Name: TektonClusterName}, cluster)
+	if err != nil {
+		reqLogger.Info("requeue to retrieve cluster resource", "tektonclustername", TektonClusterName)
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, nil
+	}
+
+	switch getTektonClusterStatus(cluster) {
+	case config.InstalledStatus:
+		reqLogger.Info("done with Tekton installation")
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationSucceeded, "tekton installation succeeded")
+	case config.InstallingStatus:
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationInstalling, "tekton installation installing")
+	case config.ErrorStatus:
+		return reconcile.Result{}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationFailed, "tekton installation failed with error")
+	default:
+		return reconcile.Result{Requeue: true, RequeueAfter: 3 * time.Second}, r.statusUpdate(reqLogger, tektonInstallation, r.setStatusTektonInstallationUnknown, "tekton installation status is unknown")
```

Comment:
I'm not too sure why this is needed. i noticed that tektoninstallation gets stuck in unknown state sometimes (it was causing the e2e-tests to fail on this PR), However, if the status is installing, it moves ahead just fine to installed. i think the watcher should automatically trigger a change, but perhaps something is wrong with how the watcher is setup? any help would be appreciated.

Reply:
Your watcher doesn't work at all. You are using `c.Watch(&source.Kind{Type: &config.Config{}}, commoncontroller.MapToOwnerByLabel("", "provider"))` which watches Tekton Config with `provider` label. The Tekton config doesn't have such label. There is supposed to be only one Config resource so I think it would be safe to watch all configs here.

Reply:
ohhhhh makes sense, okay removed that.